src/ray/common/id.h

```diff
@@ -27,6 +27,86 @@ class JobID;
 /// A helper function that get the `DriverID` of the given job.
 WorkerID ComputeDriverIdFromJob(const JobID &job_id);
 
+enum class ObjectType : uint8_t {
+  PUT_OBJECT,
+  RETURN_OBJECT,
+};
+
+enum class TransportType : uint8_t {
+  STANDARD,
+  DIRECT_ACTOR_CALL,
+};
+
+namespace object_id_helper {
+
+constexpr uint8_t is_task_offset_bits = 15;
+constexpr uint8_t object_type_offset_bits = 14;
+constexpr uint8_t transport_type_offset_bits = 11;
+
+/// A helper function to set object ids flags.
+inline void SetIsTaskFlag(uint16_t *flags, bool is_task) {
+  uint16_t is_task_bits;
+  if (is_task) {
+    is_task_bits = (0x1 << is_task_offset_bits);
+  } else {
+    is_task_bits = (0x0 << is_task_offset_bits);
+  }
+  *flags = (*flags bitor is_task_bits);
+}
+
+inline bool IsTask(uint16_t flags) {
```

Comment:
This `flags` type is used often enough that I think we should probably have a typedef for it, in case it changes from `uint16_t`. How about a type called `ObjectIDFlags`?

Reply:
Nice idea!
I'd like to add an inner class `ObjectIDFlags` that we can encapsulate the operators like `SetXXXFlag()` and `GetXXXFlag`.